src/shims/unix/socket.rs

```diff
@@ -19,7 +21,11 @@ struct SocketPair {
     // gone, and trigger EPIPE as appropriate.
     writebuf: Weak<RefCell<Buffer>>,
     readbuf: Rc<RefCell<Buffer>>,
+    /// When a socketpair instance is created, two socketpair file descriptions are generated.
+    /// The peer_fd field holds a weak reference to the file description of peer socketpair.
+    peer_fd: WeakFileDescriptionRef,
     is_nonblock: bool,
+    peer_closed: bool,
```

Comment:
Can't we just check `peer_fd.upgrade().is_some()` or so? Why do we need to track `peer_closed` separately?

Reply:
Yea this works too.